sympy/solvers/ode/single.py

```diff
@@ -422,6 +418,12 @@ def _equation(self, fx, x, order):
         P, Q = self.wilds()
         return fx.diff(x) + P*fx - Q
 
+    def _verify(self, fx):
+        if self.ode_problem.order != 1:
+            return False
```

Comment:
I don't think this should be handled in `_verify`.

This should be in `SingleODESolver.matches`. Each `SingleODESolver` subclass should have an order attribute that specifies the order of ODE that it can solve. The order should either be an int or a string like `'any'` or something. That can then be checked in `SingleODESolver.matches`.

Reply:
Thanks for the suggestion, will update PR in this direction.